src/content/en/fundamentals/instant-and-offline/pwa-architecture/index.md

```diff
@@ -0,0 +1,922 @@
+
+This option presumes you can render the same way on the client and server. The reason server-side and client-side rendering is problematic is because they are typically done in different programming environments and in different languages. For websites that blend static, navigable content and app-like interactivity, this can become a huge pain.
+
+Note: If you are using a Universal JavaScript framework, the same templating code might run on both the server and the client but that is not a requirement for using the service worker and app shell model.
+
+In an app shell architecture, a server-side component should be able to treat the content separately from how it is presented on the UI. Content could be added to a HTML layout during a SSR of the page, or it could be served up on its own to be dynamically pulled in. Static content sites such as news outlets can use PWAs and so can dynamic sites such as social media or shopping. What’s important is that the app does something meaningful when offline. 
+
+### PWA Architectural Patterns {: #pwa-architectural-patterns }
+
+PWAs can be built with any architectural style (SSR, CSR, or a hybrid of the two) but service workers imply some subtle changes in how you build your application architecture. 
+
+The following patterns are known styles for building PWAs, listed in recommended order. See the [Table of Known Patterns for Building PWAs](#table-of-known-patterns-for-building-pwas) for examples of real-world businesses using each pattern. 
+
+**Application shell (SSR) + use JavaScript to fetch content once the app shell is loaded.** 
+
+SSR is optional. Your shell is likely to be highly static, but SSR provides slightly better performance in some cases.
+
+**Application shell (SSR both shell + content for entry page) + use JavaScript to fetch content for any further routes and do a "take over"**
+
+Notes:
+
+* In the future, consider a server-side render of UI with Streams for body content model (even better). See [https://jakearchibald.com/2016/streams-ftw/](https://jakearchibald.com/2016/streams-ftw/) to learn more.
+* If you are building a PWA using Polymer leveraging this pattern, then it might be worth exploring SSR of content in the Light DOM
+
+3. Server-side rendering full page (full page caching)
+
+Note: For browsers that do not support service workers, we gracefully degrade to still server-side rendering content (for example, iOS). 
+
+**Full client-side or server-side rendering the full page (full page caching, potential for JSON payload bootstrapping via server)**
+
+#### Table of Known Patterns for Building PWAs
+
+<table class="green">
```

Comment:
Try to not use colors, but instead use the DevSite styles. If there isn't a specific reason to use green here (ie, not details), then don't use green